precondition.go

```diff
@@ -0,0 +1,8 @@
+package hsm
+
+func Precondition(srv Service, expression bool, message string) {
+	if !expression {
+		srv.Logger().Panic(message)
+		panic(message)
```

Comment:
Panic is an anti pattern. Especially in a library, should be removed. 

I’m also surprised not to find error built into the library at all. Is it designed to be infallible?

Reply:
I read up that panic should be used in Go to express programming errors.  Preconditions resulting in panics should be used when someone is calling a method and not obeying the method's contract.  My tendency is to let the program crash (even server code) on programming errors to make them easy to debug and quickly fix.  Otherwise, the program is likely to have undefined behaviour and harder to debug later when it eventually crashes for other reasons or exhibits resource problems.

The framework has a minimal number of methods, all designed such that they need to handle their errors and never return error since the framework's calling code will never know how to properly handle an error.

Reply:
logger.Panic() itself panics, so the second line is redundant :)